src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/Passes/CSharpFormattingPass.cs

```diff
@@ -147,10 +147,15 @@ public async Task<ImmutableArray<TextChange>> ExecuteAsync(FormattingContext con
             else if (lineInfo.SkipNextLineIfBrace)
             {
                 // If the next line is a brace, we skip it. This is used to skip the opening brace of a class
-                // that we insert, but Roslyn settings might place on the same like as the class declaration.
+                // that we insert, but Roslyn settings might place on the same line as the class declaration,
+                // or skip the opening brace of a lambda definition we insert, but Roslyn might place it on the
+                // next line. In that case, we can't place it on the next line ourselves because Roslyn doesn't
+                // adjust the indentation of opening braces of lambdas in that scenario.
                 if (iFormatted + 1 < formattedCSharpText.Lines.Count &&
                     formattedCSharpText.Lines[iFormatted + 1] is { Span.Length: > 0 } nextLine &&
-                    nextLine.CharAt(0) == '{')
+                    nextLine.GetFirstNonWhitespaceOffset() is { } firstNonWhitespace &&
+                    nextLine.Start + firstNonWhitespace == nextLine.End - 1 &&
```

Comment:
>nextLine.Start + firstNonWhitespace == nextLine.End - 1 && [](http://example.com/codeflow?start=20&length=58)

Just checking we don't need the equivalent for the pretty similar if condition below covering original

Reply:
Yes, good call, thanks. I think I just added this because I noticed it was missing, clearly didn't look far enough down the page :)